pp_hot.c

```diff
@@ -6028,6 +6035,19 @@ Perl_leave_adjust_stacks(pTHX_ SV **from_sp, SV **to_sp, U8 gimme, int pass)
                                             |(srcflags & SVf_IVisUV));
                         }
                     }
+                    else if (NVSIZE <= IVSIZE && (srcflags & SVf_NOK)) {
+                        SET_SVANY_FOR_BODYLESS_NV(newsv);
+                        dstflags = (SVt_NV|SVf_NOK|SVp_NOK|SVs_TEMP);
+
+                        /* both src and dst are <= SVt_MV, so sv_any points to the
+                         * head; so access the head directly
+                         */
+                        assert(    &(sv->sv_u.svu_nv)
+                                == &(((XPVNV*) SvANY(sv))->xnv_u.xnv_nv));
+                        assert(    &(newsv->sv_u.svu_nv)
+                                == &(((XPVNV*) SvANY(newsv))->xnv_u.xnv_nv));
+                        newsv->sv_u.svu_nv = sv->sv_u.svu_nv;
+                    }
```

Comment:
Alas, this needs to use the preprocessor, since for long double or quadmath builds NV is 12-16 bytes and IV is 8 bytes on 64-bit platforms.

Reply:
Thanks Tony. I'll revert to the original PR.

Reply:
The SvTYPE() comparison that mauke suggested is still fine, it just doesn't work for this block.

Reply:
Ok, I had been thinking about keeping the two changes consistent, but have just reverted this second one.